src/slurm_plugin/clustermgtd.py

```diff
@@ -0,0 +1,624 @@
+        )
+        self.terminate_drain_nodes = config.getboolean(
+            "clustermgtd", "terminate_drain_nodes", fallback=self.DEFAULTS.get("terminate_drain_nodes")
+        )
+        self.terminate_down_nodes = config.getboolean(
+            "clustermgtd", "terminate_down_nodes", fallback=self.DEFAULTS.get("terminate_down_nodes")
+        )
+        self.orphaned_instance_timeout = config.getint(
+            "clustermgtd", "orphaned_instance_timeout", fallback=self.DEFAULTS.get("orphaned_instance_timeout")
+        )
+
+    def _get_config(self, config_file_path):
+        """Get clustermgtd configuration."""
+        log.info("Reading %s", config_file_path)
+        config = ConfigParser()
+        try:
+            config.read_file(open(config_file_path, "r"))
+        except IOError:
+            log.error(f"Cannot read cluster manager configuration file: {config_file_path}")
+            raise
+
+        # Get config settings
+        self._get_basic_config(config)
+        self._get_health_check_config(config)
+        self._get_launch_config(config)
+        self._get_terminate_config(config)
+
+        # Log configuration
+        log.info(self.__repr__())
+
+
+class ClusterManager:
+    """Class for all cluster management related actions."""
+
+    class HealthCheckTypes(Enum):
+        """Enum for health check types."""
+
+        scheduled_event = "scheduled_events_check"
+        ec2_health = "ec2_health_check"
+
+        def __str__(self):
+            return self.value
+
+    def __init__(self):
+        """
+        Initialize ClusterManager.
+
+        self.static_nodes_in_replacement is persistent across multiple iteration of manage_cluster
+        This state is required because we need to ignore static nodes that might have long bootstrap time
+        """
+        self.static_nodes_in_replacement = set()
+
+    def _set_static_nodes_in_replacement(self, node_list):
+        """Setter function used for testing purposes only; not used in production."""
+        self.static_nodes_in_replacement = set(node_list)
+
+    def set_current_time(self, current_time):
+        self.current_time = current_time
+
+    def set_sync_config(self, sync_config):
+        self.sync_config = sync_config
+
+    def initialize_instance_manager(self):
+        """Initialize instance manager class that will be used to launch/terminate/describe instances."""
+        self.instance_manager = InstanceManager(
+            self.sync_config.region, self.sync_config.cluster_name, self.sync_config.boto3_config
+        )
+
+    def manage_cluster(self, sync_config):
+        """Manage cluster by syncing scheduler states with EC2 states and performing node maintenance actions."""
+        # Initialization
+        self.set_sync_config(sync_config)
+        self.set_current_time(datetime.now(tz=timezone.utc))
+        self.initialize_instance_manager()
```

Comment:
sorry if I asked already. Can this be initialized in the constructor?

Reply:
gotcha, then probably the setter for the sync_config should trigger a re-initialzation of the instance_manager.